recipe/meta.yaml

```diff
@@ -1,50 +1,69 @@
-{% set version = "1.4.19" %}
+{% set version = "2.0.0" %}
 
 package:
   name: menuinst
   version: {{ version }}
 
 source:
   url: https://github.com/conda/menuinst/archive/{{ version }}.tar.gz
-  sha256: f93ef4bc4e627868b7bba62ef8cb675e84d5bbe69a85a03a7559e1979f3924ec
+  sha256: 0ca844379c0d904cc1879f574c4d4320e89290e63e2f61e2996aebdd579d539e
 
 build:
-  number: 1
-  skip: true  # [not win]
+  number: 0
+  script_env:
+    - SETUPTOOLS_SCM_PRETEND_VERSION={{ version }}
   script:
+    # Apparently these files make the post-build linkage analysis crash
+    # and we should not need them on Windows
+    - del /q "{{ SRC_DIR }}\\menuinst\\data\\osx_launcher_*"  # [win]
+    - del /q "{{ SRC_DIR }}\\menuinst\\data\\appkit_launcher_*"  # [win]
     - {{ PYTHON }} -m pip install . -vv
-    - copy "%SRC_DIR%\\cwp.py" "%PREFIX%\\"
-  entry_points:
-    - menuinst = menuinst.main:main
-  skip_compile_pyc:
-    - cwp.py
+    # menuinst v1 expects cwp.py in PREFIX; backwards compat fix
+    - copy "%SP_DIR%\\menuinst\\_legacy\cwp.py" "%PREFIX%\\"  # [win]
+  skip_compile_pyc:  # [win]
+    - cwp.py  # [win]
+  # entry_points:
+  #   - menuinst = menuinst.main:main
 
 requirements:
+  build:
+    - python                                 # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+    - {{ compiler('c') }}  # [win]
+    - {{ compiler('cxx') }}  # [win]
   host:
     - python
     - pip
+    - setuptools_scm
   run:
     - python
 
 test:
   imports:
     - menuinst
-
+    - menuinst.api
+    - menuinst._legacy  # [win]
+    - menuinst.platforms.win_utils.winshortcut  # [win]
   commands:
-    - menuinst --help
+    - set pyver=$(python -c "import sys; print('.'.join(map(str, sys.version_info[:2])))")  # [osx]
+    - test -f "${CONDA_PREFIX}/lib/python${pyver}/menuinst/data/appkit_launcher_arm64       # [osx]
+    - test -f "${CONDA_PREFIX}/lib/python${pyver}/menuinst/data/appkit_launcher_x86_64      # [osx]
```

Comment:
These Swift binaries are compiled with [this script](https://github.com/conda/menuinst/blob/e992e76/src/appkit-launcher/build.sh), using [this config](https://github.com/conda/menuinst/blob/e992e76e4508bb58f798d52320b22936237c7e2d/src/appkit-launcher/Package.swift) which requires 10.6+.

However, the Swift runtime is only guaranteed to be present in macOS 10.14.4 and above. In the previous versions, users are supposed to install the libraries via this [DMG package](https://support.apple.com/kb/DL1998?locale=en_US).


I checked the [license](https://github.com/conda-forge/menuinst-feedstock/files/12611850/License.txt) and it doesn't look like it's distributable, so packaging as we do with Windows C runtimes might not be feasible.

Requiring `__osx>=10.14.4` is not ideal either because `conda` will depend on this package.

These launchers are only needed for an arguably niche feature (extension and protocol association) that is not in widespread usage (it's only available in the CEP-11 menuinst JSON files), and most users should be on more recent macOS versions anyway.

My question for @conda-forge/core is: are you ok with this situation? If not, what are the alternatives? Just document this limitation?

Reply:
> Requiring `__osx>=10.14.4` is not ideal either because `conda` will depend on this package.

People stay on old conda clients for longer than we'd like to already; but already-existing conda versions will continue to work with packages we produce for quite a while yet. In particular, what are the chances that the (miniscule amount of) users still on ancient MacOS are running the latest conda versions? In practice, I doubt that it matters one bit if those users now have a for-real cap on their max. conda. I'd say go ahead.

Reply:
I think I found a compromise: to document this in the `menuinst` docs, saying that if you are using protocol/MIME association on macOS, you should add `__osx>=10.14.4` to the package containing the menu JSON document, or tell their users to install the libraries.

Thanks for the feedback!

Reply:
https://github.com/conda/menuinst/pull/160